gui/kitchen-info.lua

```diff
@@ -0,0 +1,404 @@
+    }
+}
+
+local COLUMNS_BY_LABEL = {}
+
+for _, column in ipairs(COLUMNS) do
+    column.activeOnPages = utils.invert(column.activeOnPages)
+    column.minLength = math.max(#column.label, column.minValueLeftPadding + column.maxValueLength)
+
+    COLUMNS_BY_LABEL[column.label] = column
+end
+
+function COLUMNS_BY_LABEL.Extract.getColumnValue(item_type, material, plant_raw)
+    if (item_type == df.item_type.PLANT) then
+        if (plant_raw.flags.EXTRACT_BARREL) then
+            return "Barrel"
+        elseif (plant_raw.flags.EXTRACT_VIAL) then
+            return " Vial"
+        elseif (plant_raw.flags.EXTRACT_STILL_VIAL) then
+            return " Still"
+        end
+    end
+    return false
+end
+
+-- This column is hidden because the only plant with BAG_ITEM reaction is Quarry Bush, which is neither cookable nor brewable so it never appears in the Kitchen screen
+function COLUMNS_BY_LABEL.ToBag.getColumnValue(item_type, material, plant_raw)
+    if (item_type == df.item_type.PLANT) then
+        local reaction_product = material.reaction_product
+        for i, reaction_product_id in ipairs(reaction_product.id) do
+            if (reaction_product_id.value == "BAG_ITEM") then
+                local bag_item_type = reaction_product.item_type[i]
+
+                if (bag_item_type == df.item_type.PLANT_GROWTH) then
+                    local bag_item_subtype = reaction_product.item_subtype[i]
+                    local bag_mat_type  = reaction_product.material.mat_type[i]
+                    local bag_mat_index = reaction_product.material.mat_index[i]
+                    local bag_matinfo = dfhack.matinfo.decode(bag_mat_type, bag_mat_index)
+                    local bag_mat_plant_raw = bag_matinfo.plant
+                    local bag_mat_plant_growth = bag_mat_plant_raw.growths[bag_item_subtype]
+
+                    if (bag_mat_plant_growth.name_plural:sub(-6) == "leaves") then
+                        return "Leaves"
+                    end
+                end
+
+                return "Other"
+            end
+        end
+    end
+    return false
+end
+
+function COLUMNS_BY_LABEL.Thread.getColumnValue(item_type, material, plant_raw)
+    if (item_type == df.item_type.PLANT and plant_raw.flags.THREAD) then
+        return "Thread"
+    end
+    return false
+end
+
+function COLUMNS_BY_LABEL.Mill.getColumnValue(item_type, material, plant_raw)
+    if (item_type == df.item_type.PLANT and plant_raw.flags.MILL) then
+        local mill_mat_type = plant_raw.material_defs.type.mill
+        local mill_mat_index = plant_raw.material_defs.idx.mill
+        local mill_matinfo = dfhack.matinfo.decode(mill_mat_type, mill_mat_index)
+        local mill_material = mill_matinfo.material
+
+        if (mill_material.flags.IS_DYE) then
+            return "Dye"
+        elseif (mill_material.state_name.Powder:sub(-5) == "flour") then
+            return "Flour"
+        else
+            return "Other"
+        end
```

Comment:
What's the significance here of differentiating "flours" from others?

I'm very leery of using substring matches into item names or descriptions; just because it yields the desired results on vanilla items doesn't mean it always will, or that it will do so on modded items as well.

Reply:
The point is user friendlyness.
I am already differenting Dye by the flag.
For the rest when the user sees Flour he will know that by milling Plant Abc he will get Plant Abc Flour, like any other flour in the game.
When he sees Other, he will have to look up what sort of Powder is created and whats its use.

If that doesn't give desired results for some modded Flour, which despite name is not really a Flour, then it is the problem of user-friendlyness of mod that named it.
Yes it is sort of hardcoding vanilla stuff, but most thing on this screen relly on assumptions from vanilla, as many jobs like "Process to bag" or "Mill to paste" are a certain vanilla reaction with certain vanilla tags on items.